src/analyses/mCP.ml

```diff
@@ -268,6 +268,9 @@ struct
           Result.meet a @@ S.query ctx' q
         in
         match q with
+        | Queries.Invariant context ->
+          let e = D.invariant context ctx.local in
+          BatOption.map_default (Queries.ES.singleton) (Queries.ES.top ()) e
```

Comment:
Instead of implementing this in `MCP`, just implement it at the very outer level using `Spec.D.invariant`, just like witnesses do it. Then all the existing path-joining logic is reused.

Reply:
yes, that seems to do the trick!